### ISSUE TITLE
Library not working with Swift 3

### ISSUE THREAD

**Issue by Umang0609** [IBM-Swift/Swift-JWT#4]
Hi. Does this library work with Swift 3? Can you provide Pod for the same

**Comment by ShihabMehboob** [comment]
Pod has now been created, to use it, simply add the following to your Pod file:

pod 'SwiftJWT', :git => 'https://github.com/IBM-Swift/Swift-JWT', :branch => 'Pod'

**Comment by KyeMaloy97** [comment]
As for Swift 3 support, as it is an entirely new repo we have launched it only supporting Swift 4 and above. I will look into adding Swift 3.1.1 support though 😄

**Comment by Umang0609** [comment]
@ShihabMehboob Thanks a lot for this kind support . I tried to use this SwiftJWT Pod using Swift 3.2 but there are lot are errors in Blue RSA . Could you please help to check 
<img width="1764" alt="screen shot 2018-06-07 at 10 27 13 am" src="https://user-images.githubusercontent.com/36652918/41075108-b5c9ca56-6a3d-11e8-8525-5da86408f3a1.png">
<img width="1764" alt="screen shot 2018-06-07 at 10 27 31 am" src="https://user-images.githubusercontent.com/36652918/41075109-b5f47eea-6a3d-11e8-85fc-664a183b45ac.png">

**Comment by Umang0609** [comment]
Hi, I managed to collect the missing classes from CommonCrypto and able to run my code now. I am trying to encode the JWT with the help of your Library. I am using my Custom Headers and Claims. For the Signed JWT , I have signed Certificate with me which is valid but I am getting nil in Signed JWT, below is my code - 

data is the Certificate Data and key I am using .certificate. Can you please advice how I will get the signed JWT . Any help will be greatly appreciated!

 var jwt = JWT(header: Header(["x5c" : cert,"alg" : "RS-256","exp" : expirytime]), claims: Claims(["scope": "cop"]))
                        let signedJwt = try? jwt.sign(using: .rs256(data, .certificate))

**Comment by KyeMaloy97** [comment]
Have you tried it with the dash in RSA-256 removed? Alg is a supported enum too so you can just do .alg:"rsa256", give that a try and let me know if it solves your problem :)

**Comment by Umang0609** [comment]
Yes I tried with enums too but its not returning the JWT. The Private Key and Certificate I checked online, both are valid . The Algorthim used to sign Signature is not returning the Signature due to which it returns Nil.
<img width="832" alt="screen shot 2018-06-07 at 8 09 33 pm" src="https://user-images.githubusercontent.com/36652918/41098684-d1529900-6a8e-11e8-9bb3-94f0f891f175.png">

**Comment by Umang0609** [comment]
I am using the certificate issued by CA server. Its not generated with Passphrase. First, I am creating the CSR with private key and then send it to server , server returns the valid Certificated signed with CA . Now this Certificate I am using to create JWT Token using your Library.

var jwt = JWT(header: Header([.typ:"JWT",.x5c:cert]), claims: Claims(["scope": "cop"])) // Library is automatically creating the Algorithm and add in Header

let jwtToken = try? jwt.sign(using: .rs256(cert.data(using: .utf8)!, .certificate)) 

but JWT Token is nil . Its not encoding the Signature .
Could you provide some sample app which generate JWT using Certificate ?

**Comment by KyeMaloy97** [comment]
I will have a go using a certificate to see what I can work out :)

**Comment by djones6** [comment]
Hi @Umang0609,

I am far from an expert on JWT, however from [the example given in the Swift-JWT tests](https://github.com/IBM-Swift/Swift-JWT/blob/afcd8f205a96b9c6f0a02ed01496087821d44ed8/Tests/SwiftJWTTests/TestJWT.swift#L89) it looks like you should *sign* a JWT using a private key, and *verify* a JWT using either the corresponding public key, or a certificate containing that key.

I suspect that what is happening in your case is that the certificate data you are passing in your call to `jwt.sign` is being misinterpreted as a private key, and the signing process is failing.

**Comment by Andrew-Lees11** [comment]
Hello @Umang0609, 
I would like to close out this issue and wanted to know if you able to get your SwiftJWT project working with Swift 3? 

We are about to release a new Major of SwiftJWT which will use Codable API's. This requires swift 4 and so we will no longer be supporting Swift 3 in the latest release.
